mypy/typeops.py

```diff
@@ -306,7 +306,11 @@ class B(A): pass
     self_param_type = get_proper_type(func.arg_types[0])
 
     variables: Sequence[TypeVarLikeType]
-    if func.variables and supported_self_type(self_param_type):
+    # Having a def __call__(self: Callable[...], ...) can cause infinite recursion. Although
+    # this special-casing looks not very principled, there is nothing meaningful we can infer
+    # from such definition, since it is inherently indefinitely recursive.
+    allow_callable = func.name is None or not func.name.startswith("__call__ of")
```

Comment:
`"__call__ of"` looks weird to me -- mind telling me what's going on?

Reply:
This is just how names work for methods.